src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/updateCVECandidates.hpp

```diff
@@ -0,0 +1,190 @@
+ *
+ */
+class UpdateCVECandidates final
+{
+public:
+    /**
+     * @brief Inserts the candidate data into the corresponding database.
+     *
+     * @param resource CVE ID.
+     * @param cve5Flatbuffer CVE5 Flatbuffer.
+     * @param DBMap map with rocksDB instances.
+     */
+    static void storeVulnerabilityCandidate(const std::string& resource,
+                                            const cve_v5::Entry* cve5Flatbuffer,
+                                            std::unordered_map<std::string, Utils::RocksDBWrapper>& DBMap)
+    {
+        if (!cve5Flatbuffer || !cve5Flatbuffer->containers())
+        {
+            return;
+        }
+
+        flatbuffers::FlatBufferBuilder builder;
+
+        auto candidateLambda = [&](const cve_v5::Affected* affected, const std::string& containerId)
+        {
+            if (!affected->product())
+            {
+                return;
+            }
+
+            auto cveIdFB = builder.CreateString(resource);
+
+            std::vector<flatbuffers::Offset<NSVulnerabilityScanner::Version>> versionFBArray;
+
+            if (affected->versions())
+            {
+                for (const auto& versionElement : *affected->versions())
+                {
+                    NSVulnerabilityScanner::Status status;
+                    flatbuffers::Offset<flatbuffers::String> version;
+                    flatbuffers::Offset<flatbuffers::String> lessThan;
+                    flatbuffers::Offset<flatbuffers::String> lessThanOrEqual;
+                    flatbuffers::Offset<flatbuffers::String> versionType;
+
+                    if (versionElement->status())
+                    {
+                        status = VERSION_STATUS_MAP.at(versionElement->status()->str());
+                    }
+
+                    if (versionElement->version())
+                    {
+                        version = builder.CreateString(versionElement->version()->str());
+                    }
+
+                    if (versionElement->lessThan())
+                    {
+                        lessThan = builder.CreateString(versionElement->lessThan()->str());
+                    }
+
+                    if (versionElement->lessThanOrEqual())
+                    {
+                        lessThanOrEqual = builder.CreateString(versionElement->lessThanOrEqual()->str());
+                    }
+
+                    if (versionElement->versionType())
+                    {
+                        versionType = builder.CreateString(versionElement->versionType()->str());
+                    }
+
+                    NSVulnerabilityScanner::VersionBuilder versionBuilder(builder);
+                    versionBuilder.add_status(status);
+                    versionBuilder.add_version(version);
+                    versionBuilder.add_lessThan(lessThan);
+                    versionBuilder.add_lessThanOrEqual(lessThanOrEqual);
+                    versionBuilder.add_versionType(versionType);
```

Comment:
Maybe using a call to the method "CreateVersionDirect" allows us to keep the code cleaner